code/ai/aicode.cpp

```diff
@@ -435,6 +435,9 @@ void garbage_collect_path_points()
 	//	in pp_xlate, mark all used Path_point records
 	for ( so = GET_FIRST(&Ship_obj_list); so != END_OF_LIST(&Ship_obj_list); so = GET_NEXT(so) ) {
 		A = &Objects[so->objnum];
+		if (A->flags[Object::Object_Flags::Should_be_dead])
+			continue;
+
```

Comment:
I took a quick look at ship cleanup, and these paths won't get cleaned up if a ship was vanished while exiting out of a bay.  I think this would barely happen, but I think this points to the fact that we really should vectorize or maybe refactor path points.  There's no reason to limit the number, and this function probably doesn't need to exist.

Reply:
Can you elaborate on the connection between ship cleanup and path garbage collection?  The `garbage_collect_path_points` function is only called twice, both when paths are created, as opposed to when ships exit.  It also does not seem to depend in any way on whether a ship is vanished or not.

Reply:
I mean it's just an issue with being able to create enough path points.  If, in the course of creating path points, a ship vanished this frame holds many path points, this check may keep enough of them from being freed for the ships being created.  I mean, it's an extreme edge case and not likely to happen.  I just think someone should make path points dynamic at some point and clean up those points when the ship departs.

Reply:
I looked over the functions again and the path points should be properly cleaned up any time they are needed, whether a ship is vanished or not.  Vectorizing the list might be a good idea, but that's a separate issue from this PR.  In the meantime, 1000 is high enough that I don't think it's likely to be hit, in any case.

Reply:
Agreed